ddtrace/internal/telemetry/writer.py

```diff
@@ -77,21 +78,26 @@ def _send_request(self, request):
                 conn.close()
 
     def _flush_integrations_queue(self):
-        # type () -> List[Dict]
+        # type: () -> List[Dict]
         """Returns a list of all integrations queued by add_integration"""
         with self._lock:
             integrations = self._integrations_queue
             self._integrations_queue = []
         return integrations
 
     def _flush_events_queue(self):
-        # type () -> List[Dict]
+        # type: () -> List[Dict]
         """Returns a list of all integrations queued by classmethods"""
         with self._lock:
             requests = self._events_queue
             self._events_queue = []
         return requests
 
+    def _reset_queues(self):
+        # type: () -> None
+        self._integrations_queue = []
+        self._events_queue = []
```

Comment:
Do we need a lock here?

Reply:
Initially I thought no. Assigning a value to an array is an atomic operation so it doesn't need a lock but this could lead to issues down the line. Adding a lock here would make the telemetry writer more maintainable